### DISCUSSION TITLE
[Web3Modal - wagmi  ethers adapter] - How can I listen to provider events

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#2653] [Q&A] [answered]
I used an adapter for the signer and provider as [described here](https://wagmi.sh/react/ethers-adapters), but how can I listen to it?
What I want to do is refresh the page and make an API call when the chain changes.

**Comment by kehiy** [comment, marked as answer]
for To listen to provider events using the Web3Modal library with the wagmi ethers adapter
just import necessary dependencies:
```js
import { ethers } from 'ethers';
import { ProviderEvents } from 'web3modal';
```
create instance of ethers.js and signer

listen the events:
```js
provider.on(ProviderEvents.ChainChanged, (chainId) => {
  // your code
});

provider.on(ProviderEvents.AccountChanged, (accounts) => {
 // your code
});
```

**Reply by gkfyr** [reply]
Hi! is this still available at ethers v6?

**Reply by kehiy** [reply]
Hi, I'm not sure. you can test.
